### PR TITLE
Add some documentation to overallStatus()

### PR DESCRIPTION
**Pull request by fejta**
/cc @chases2 

I am concerned with doing this sort of thing, because people tend to be more fluent in English than golang. Thus minds prefer reading the English. But only the golang determines how the program actually behaves. We need people maintaining this code to read the golang.

### CODE REVIEW

**Comment by fejta** [comment]
/retest 

flake https://github.com/GoogleCloudPlatform/testgrid/issues/466

File: pkg/summarizer/summary.go [hunk 1]
```diff
@@ -571,6 +571,13 @@ func buildFailLink(testID, target string) string {
 }
 
 // overallStatus determines whether the tab is stale, failing, flaky or healthy.
+//
+// Tabs are:
```
Comment (chases2) [inline pkg/summarizer/summary.go:575]: In this example, this helps a whole lot. It tells me that you're trying to cover for 5 cases, and a bit about what each one means. Then, looking at the function, I see that 3 of the 5 are handled immediately, so most of the function is looking for failing cells in recent columns. Nice.

If I'm looking to see what "Flaky" means, I _might_ stop here, and that's maybe a risk, but I can also dig into the code to figure out how "recent" works or if I think it's broken.

**Review by chases2 — APPROVED** [review]
> /cc @chases2
> 
> I am concerned with doing this sort of thing, because people tend to be more fluent in English than golang. Thus minds prefer reading the English. But only the golang determines how the program actually behaves. We need people maintaining this code to read the golang.

Reading the golang is made easier when there's some indication of what the function or author was trying to do. Otherwise, it's tough to judge if code is overengineered, or not sufficient, or bugged.

File: pkg/summarizer/summary.go (hunk 1 above)
Comment (fejta) [reply pkg/summarizer/summary.go:575]: The thing is, I do not want anyone maintaining this code who "stops there." Maintainers need to read the code, and documentation makes it easier for people to not do this.

**Pull request merged**